auto-self-nominate/src/AccountState.ts

```diff
@@ -0,0 +1,100 @@
+import { SDK } from "codechain-sdk";
+import { PlatformAddress } from "codechain-sdk/lib/core/classes";
+
+const RLP = require("rlp");
+
+export type AccountState = "Banned" | "Candidates" | "Jailed";
+export interface JailedAccountInfo {
+    deposits: number;
+    custodyUntil: number;
+    releasedAt: number;
+}
+export interface CandidatesInfo {
+    [key: string]: {
+        deposits: number;
+        nominateEndAt: number;
+    };
+}
+export interface BannedAccountsInfo {
+    accounts: string[];
+}
+export interface JailedAccountsInfo {
+    [key: string]: {
+        deposits: number;
+        custodyUntil: number;
+        releasedAt: number;
+    };
+}
+
+export type StatesInfos = CandidatesInfo | BannedAccountsInfo | JailedAccountsInfo;
+
+function decodeNumber(buffer: Buffer): number {
```

Comment:
I think `decodeInt` would be a better name.

Reply:
I renamed it.